test/check_an_util.c

```diff
@@ -0,0 +1,137 @@
+#include <assert.h>
+#include <check.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "an_util.h"
+
+static bool
+is_power_of_2(uint64_t x)
+{
+	return ((x != 0) && !(x & (x - 1)));
+}
+
+START_TEST(test_an_next_power_of_2)
+{
+
+	for (uint64_t i = 1; i < UINT16_MAX; i += 15) {
```

Comment:
why can't we do all ints? is it that slow? If so, can we do exhaustive up to, e.g, 1024?

Reply:
We can probably do exhaustive until INT16_MAX.  I will push that change